exo-go/src/app_dependency_helpers/generic.go

```diff
@@ -16,3 +29,48 @@ func (dependency genericDependency) GetDeploymentConfig() map[string]string {
 	}
 	return config
 }
+
+// GetDockerConfig returns docker configuration for the dependency and an error if any
+func (dependency genericDependency) GetDockerConfig() (types.DockerConfig, error) {
+	renderedVolumes, err := dependency.getRenderedVolumes()
+	if err != nil {
+		return types.DockerConfig{}, err
+	}
+	return types.DockerConfig{
+		Image:         fmt.Sprintf("%s:%s", dependency.config.Name, dependency.config.Version),
+		ContainerName: dependency.GetContainerName(),
+		Ports:         dependency.config.Config.Ports,
+		Volumes:       renderedVolumes,
+	}, nil
+}
+
+// GetEnvVariables returns the environment variables for the depedency
+func (dependency genericDependency) GetEnvVariables() map[string]string {
+	return dependency.config.Config.DependencyEnvironment
+}
+
+// GetOnlineText returns the online text for the dependency
+func (dependency genericDependency) GetOnlineText() string {
+	return dependency.config.Config.OnlineText
+}
+
+func (dependency genericDependency) getRenderedVolumes() ([]string, error) {
+	homeDir, err := osHelpers.GetUserHomeDir()
+	if err != nil {
+		return []string{}, err
+	}
```

Comment:
Thoughts on making the homeDir a property of genericDependency? This would be passed in when its created and would thus need to be an argument to `appDependencyHelper.Build`. That would make this easier to test.